cumulus/tasks/queue-pdrs/tests/index.js

```diff
@@ -1,53 +1,170 @@
+    { name: randomString(), path: randomString() }
+  ];
+
+  await validateConfig(t, event.config);
+  await validateInput(t, event.input);
 
-test('queue pdrs', async (t) => {
-  const Bucket = t.context.bucket;
-  const ParsePdrTemplate = `s3://${Bucket}/dev/workflows/ParsePdr.json`;
+  const output = await queuePdrs(event);
 
-  await aws.s3().putObject({
-    Bucket,
-    Key: 'dev/workflows/ParsePdr.json',
-    Body: JSON.stringify(workflowTemplate)
+  await validateOutput(t, output);
+
+  // Get messages from the queue
+  const receiveMessageResponse = await sqs().receiveMessage({
+    QueueUrl: t.context.event.config.queueUrl,
+    MaxNumberOfMessages: 10,
+    WaitTimeSeconds: 1
   }).promise();
+  const messages = receiveMessageResponse.Messages;
 
-  MockAWS.stub('StepFunctions', 'describeExecution').returns({
-    promise: () => Promise.resolve({})
-  });
+  t.is(messages.length, 2);
+});
+
+test('The correct message is enqueued', async (t) => {
+  const event = t.context.event;
+  event.input.pdrs = [
+    {
+      name: randomString(),
+      path: randomString()
+    },
+    {
+      name: randomString(),
+      path: randomString()
+    },
+  ];
+
+  await validateConfig(t, event.config);
+  await validateInput(t, event.input);
+
+  const output = await queuePdrs(event);
 
-  const input = Object.assign({}, inputJSON);
-  input.config.templateUri = ParsePdrTemplate;
-  input.config.queueUrl = t.context.queueUrl;
+  await validateOutput(t, output);
 
-  return handler(input, {}, (e, output) => {
-    t.ifError(e);
-    t.is(typeof output, 'object');
-    t.is(output.pdrs_queued, 2);
-    MockAWS.StepFunctions.restore();
+  // Get messages from the queue
+  const receiveMessageResponse = await sqs().receiveMessage({
+    QueueUrl: t.context.event.config.queueUrl,
+    MaxNumberOfMessages: 10,
+    WaitTimeSeconds: 1
+  }).promise();
+  const messages = receiveMessageResponse.Messages.map((message) => JSON.parse(message.Body));
+
+  t.is(messages.length, 2);
+
+  const receivedPdrnames = messages.map((message) => message.payload.pdr.name);
+  event.input.pdrs.map((pdr) => pdr.name).forEach((pdrName) =>
+    t.true(receivedPdrnames.includes(pdrName)));
+
+  // Figure out what messages we should have received for each PDR
+  const expectedMessages = {};
+  event.input.pdrs.forEach((pdr) => {
+    expectedMessages[pdr.name] = {
+      cumulus_meta: {
+        state_machine: t.context.stateMachineArn
+      },
+      meta: {
+        collection: { name: 'collection-name' },
+        provider: { name: 'provider-name' }
+      },
+      payload: {
+        pdr: {
+          name: pdr.name,
+          path: pdr.path
+        }
+      }
+    };
+  });
+
+  // Make sure we did receive those messages
+  messages.forEach((message) => {
+    const pdrName = message.payload.pdr.name;
+    t.deepEqual(message, expectedMessages[pdrName]);
   });
 });
+
+test.todo('An appropriate error is thrown if the message template could not be fetched');
```

Comment:
Is there a ticket for this?

Reply:
There is now - https://bugs.earthdata.nasa.gov/browse/CUMULUS-368